src/classes/OPP_StageMappingUtil.cls

```diff
@@ -91,10 +95,180 @@ public with sharing class OPP_StageMappingUtil {
+
+        return md;
     }
 
+    /*******************************************************************************************************
+    * @description Deploys the Custom metadata records using the OppStageToStateMetadataHandler as the handler
+    ********************************************************************************************************/
+    public static void deployOppStageToStateMetadata(List<Metadata.CustomMetadata> listOfMetadataToInsert, Map<String, Object> params) {
+        // Deploy the metadata 
+        String jobId = CMT_MetadataAPI.getUniqueJobId();
+        CMT_MetadataAPI.MetadataCallbackHandler oppCallbackHandler = new OppStageToStateMetadataHandler();
+        CMT_MetadataAPI.MetadataCallBack callback = new CMT_MetadataAPI.MetadataCallBack(jobId);
+        callback.callbackHandler = oppCallbackHandler;
+        callback.callbackParams = params;
+
+        CMT_MetadataAPI.deployMetadata(listOfMetadataToInsert, callback);
+    }
+
+    public class OppStageToStateMetadataHandler implements CMT_MetadataAPI.MetadataCallbackHandler {
+
+        public Map<String, Object> params;
+
+
+        /**************************************************************************************************************
+        * @description success handler - No implementation as of right now, will be implemented later as needed. 
+        /*************************************************************************************************************/
+        public void performSuccessHandler(Map<String, Object> params, String status) {
+            if (params == null || params.isEmpty()) {
+                return;
+            }
+
+            try {
+                // TODO: Implement if needed. 
+                // Process a successful 
+            } catch (Exception ex) {
+                logError(ex);
+            }
+        }
+
+        /**************************************************************************************************************
+        * @description error handler - No implementation as of right now, will be implemented later as needed. 
+        /*************************************************************************************************************/
+        public void performErrorHandler(Map<String, Object> params, String status) {
+            if (params == null || params.isEmpty()) {
+                return;
+            }
+
+            try {
+                // TODO: Implement if needed. 
+                // Handle error 
+            } catch (Exception ex) {
+                logError(ex);
+            }
+        }
+
+
+        /**
+        * @description Inserts an error log into the Error table
+        * @param ex Exception object that is to be logged
+        */
+        private void logError(Exception ex) {
+            Error__c rollupHandlerError = new Error__c(
+                Error_Type__c = 'Opportunity Stage to State Metadata Handler Error',
```

Comment:
If anyone will ever see this, it will need to be added to the Catalogue section of the Work Item, and converted to a Label.

Reply:
As of this moment, it should never be shown. Should we preemptively convert it into a label?

Reply:
Lets convert it.  One less thing for later.

Reply:
Actually, this is for the "type" of error, does it really need to be a label? Also, will users typically look at the error log or is that primarily for internal usage?